### PR TITLE
Making sot-dynamic-pinocchio appear as sot-dynamic-pinocchio

### PR DESCRIPTION
**Pull request by olivier-stasse**
Right now sot-dynamic-pinocchio is disguising itself as sot-dynamic.
This is a nice as long as we have only one dynamic library used in the SoT.
As it is not sure that we will have the same API between various dynamics library and that this pose problem in terms of naming if pc config files, I suggest that we move to a naming convention making clear which dynamics library is used.

### CODE REVIEW

File: src/dynamic.cpp [hunk 1]
```diff
@@ -28,10 +28,10 @@ extern "C" {
   ::dynamicgraph::Entity*						
   EntityMaker_Dynamic(const std::string& objname)
```
Comment (proyan) [inline src/dynamic.cpp]: This should be changed to EntityMaker_DynamicPinocchio as the new entity name should match Class Type

File: src/dynamic.cpp [hunk 2]
```diff
@@ -28,10 +28,10 @@ extern "C" {
   ::dynamicgraph::Entity*						
   EntityMaker_Dynamic(const std::string& objname)			
   {									
-    return new Dynamic (objname);					
+    return new DynamicPinocchio (objname);					
   }									
   ::dynamicgraph::EntityRegisterer					
-  reg_Dynamic ("Dynamic",						
+  reg_Dynamic ("DynamicPinocchio",						
 	       &EntityMaker_Dynamic);
```
Comment (proyan) [inline src/dynamic.cpp]: It should be EntityMaker_DynamicPinocchio as well

**Review by proyan — CHANGES_REQUESTED** [review]
Could you also please do these changes in the EntityMaker?

**Comment by olivier-stasse** [comment]
The new commit implements your request.

**Comment by proyan** [comment]
Thanks.

**Pull request merged**